app/auth/forms.py

```diff
@@ -78,4 +78,10 @@ class ResetPasswordRequestForm(FlaskForm):
 class ResetPasswordForm(FlaskForm):
     password = PasswordField('Password', validators=[DataRequired()])
     password2 = PasswordField('Repeat Password', validators=[DataRequired(), EqualTo('password')])
-    submit = SubmitField('Password Reset')
\ No newline at end of file
+    submit = SubmitField('Password Reset')
+
+class ResetPasswordFormPreferences(FlaskForm):
+    oldPassword = PasswordField('Old Password', validators=[DataRequired()])
+    password = PasswordField('New Password', validators=[DataRequired()])
+    password2 = PasswordField('Repeat New Password', validators=[DataRequired(), EqualTo('password')])
+    submit = SubmitField('Password Reset')
```

Comment:
I would put the old password validation here within the class.

Here is the link on how to make the validation.
https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-v-user-logins

It's the third or fourth paragraph in the User Registration section.

search for "validate_<field_name>"

Reply:
For the submitField, should we change it to "Change Password" vs "Password Reset"?

Reply:
I am able to set up a validation for the old password, but I am having trouble with passing the current user into the form, since I have to use the check_password method to see if the password they submitted is their correct password. Any Ideas on how to do this?